### PR TITLE
fix: scalar generation when scalar is taken as input

### PR DESCRIPTION
**Pull request by ssddOnTop**
fix generation of input type when scalar is used as input.

## Summary by CodeRabbit

- **New Features**
	- Introduced a `JSON` scalar type to enhance data structure flexibility.
- **Refactor**
	- Improved argument filtering in configuration settings based on scalar types.
	- Enhanced field type validation logic to ensure compliance with predefined scalar types.
- **Bug Fixes**
	- Fixed issues related to scalar type validation in various modules.
- **Tests**
	- Added a new test function to verify GraphQL schema parsing and conversion accurately.
- **Chores**
	- Removed the `KeyValues` scalar type to streamline data representation options.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -80,6 +80,11 @@ pub fn is_default<T: Default + Eq>(val: &T) -> bool {
     *val == T::default()
 }
 
+pub fn is_scalar(type_name: &str) -> bool {
+    ["String", "Int", "Float", "Boolean", "ID", "JSON"].contains(&type_name)
+        || scalars::CUSTOM_SCALARS.keys().any(|v| (*v).eq(type_name))
+}
```
[the remark this answers is not in the document]
Comment (tusharmath) [reply src/lib.rs:86]: @coderrabbitai suggest a more performant implementation

[the remark this answers is not in the document]
Comment (tusharmath) [reply src/lib.rs:86]: @coderabbitai suggest a more performant implementation

[the remark this answers is not in the document]
Comment (tusharmath) [reply src/lib.rs:86]: Nice, create a patch. Also move this function to src/scalar/mod.rs @coderabbitai

File: src/scalars/mod.rs [hunk 2]
```diff
@@ -1,4 +1,4 @@
-use std::collections::HashMap;
+use std::collections::{HashMap, HashSet};
 use std::sync::Arc;
 
 use async_graphql_value::ConstValue;
```
[the remark this answers is not in the document]
Comment (ssddOnTop) [reply src/scalars/mod.rs:4]: that is intentional, marking this as resolved

File: src/scalars/mod.rs [hunk 3]
```diff
@@ -9,12 +9,20 @@ pub use crate::scalars::email::Email;
...
     };
 }
+lazy_static! {
+    pub static ref SCALAR_TYPES: HashSet<&'static str> = {
+        let mut set = HashSet::new();
+        set.extend(["String", "Int", "Float", "Boolean", "ID", "JSON"]);
+        set.extend(CUSTOM_SCALARS.keys().map(|k| k.as_str()));
+        set
+    };
+}
```
Comment (tusharmath) [inline src/scalars/mod.rs:25]: Keep `SCALAR_TYPES` private and expose a `is_scalar` function.

**Review by tusharmath — APPROVED** [review]

**Review by tusharmath — APPROVED** [review]

**Pull request merged**